resource/authentication_manager.go

```diff
@@ -38,16 +43,56 @@ func (a *AuthenticationManager) CreateCredential(key string, id string, ver stri
 		return errors.New("unable to generate new authentication token")
 	}
 
+	currUser, err := user.Current()
```

Comment:
Need comment here explaining the problem and how this fixes it. Line 87 isnt enough.